File: majority_element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int, int> mapint;
        for (auto it = nums.begin(); it != nums.end(); ++it){
            mapint[*it]++;
        }
        int result = 0, resnum = 0;
        for (auto it = mapint.begin(); it != mapint.end(); ++it){
            if (it->second > resnum){
                resnum = it->second;
                result = it->first;
            }
        }
        return result;
    }
};

File: python/majority_element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        for i in set(nums):
            if nums.count(i) > len(nums) / 2:
                return i
        """
        dic = {}
        l = len(nums)
        l = l / 2
        for n in nums:
            dic.setdefault(n, 0)
            dic[n] += 1
        for key in dic:
            if dic[key] >= l:
                return key
        """
